network/network_test.go

```diff
@@ -0,0 +1,79 @@
+package network
+
+import (
+	"reflect"
+	"testing"
+)
+
+func Test_calculateNeighbours(t *testing.T) {
+	tests := []struct {
+		n                  int
+		ip                 string
+		addresses          []string
+		expectedNeighbours []string
+	}{
+		{
+			n:                  1,
+			ip:                 "127.0.0.1",
+			addresses:          []string{},
+			expectedNeighbours: []string{},
+		},
+		{
+			n:                  1,
+			ip:                 "127.0.0.1",
+			addresses:          []string{"127.0.0.1"},
+			expectedNeighbours: []string{"127.0.0.1"},
```

Comment:
Shouldn't this be rather `[]string{}`?

Reply:
Realistically, it's a degenerate case. However, I would argue that in a ring of one, you are your own neighbour.

Reply:
Agreed. Then this suggestion is valid https://github.com/giantswarm/net-exporter/pull/66/commits/b03e420f941dffa0c6bd2e85ed809172e787aa4a#r340466365

Reply:
My aim was to avoid having some quick return for the degenerate cases - I find I trust the algorithm a lot more like that.